### PR TITLE
Fix incorrect argument handling for debounce and throttle.

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes https://github.com/BackburnerJS/backburner.js/issues/296

### CODE REVIEW

**Comment by krisselden** [comment]
Don’t mix args and arguments just be consistent with args

**Comment by rwjblue** [comment]
Thanks for reviewing @krisselden! Updated to use `args` consistently...

File: lib/index.ts [hunk 1]
```diff
@@ -372,22 +374,73 @@ export default class Backburner {
...
+    let target;
+    let method;
+    let immediate;
     let isImmediate;
     let wait;
 
-    if (isCoercableNumber(immediate)) {
-      wait = immediate;
+    if (args.length === 1) {
+      method = targetOrThisArgOrMethod;
```
Comment (bekzod) [inline lib/index.ts]: maybe just name it `targetOrMethod` ?

File: lib/index.ts [hunk 2]
```diff
@@ -372,22 +374,73 @@ export default class Backburner {
...
+      wait = args[0];
```
Comment (bekzod) [inline lib/index.ts]: I think it should `args.pop()` here, else timer `wait` will be passed to debounced function

File: lib/index.ts [hunk 3]
```diff
@@ -415,22 +468,73 @@ export default class Backburner {
...
```
Comment (bekzod) [inline lib/index.ts]: I think it should args.pop() here, else timer wait will be passed to debounced function

File: lib/index.ts (hunk 1 above)
Comment (rwjblue) [reply lib/index.ts]: I kinda like the name 🗡

File: lib/index.ts (hunk 2 above)
Comment (rwjblue) [reply lib/index.ts]: Yep, you are definitely correct.  I updated to fix that and ensure it is under test...

**Pull request merged**